tests/e2e/e2e_hashivault_test.go

```diff
@@ -0,0 +1,110 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Simple Client-Server pod test using Vault", func() {
+	Context("SimpleClientServer", func() {
+		sourceNs := "client"
+		destNs := "server"
+		var ns []string = []string{sourceNs, destNs}
+
+		It("Tests HTTP traffic for a simple client-server pod deployment", func() {
+			// Install OSM
+			installOpts := td.GetTestInstallOpts()
+			installOpts.certManager = "vault"
```

Comment:
would there... be a way to make `vault` a flag? so basically any test can run with vault?

making vault a flag of the parseFlags thing?